sdk/all/src/main/java/io/opentelemetry/sdk/OpenTelemetrySdk.java

```diff
@@ -32,18 +42,24 @@ public static OpenTelemetrySdkBuilder builder() {
 
   /** Returns the global {@link OpenTelemetrySdk}. */
   public static OpenTelemetrySdk get() {
-    return (OpenTelemetrySdk) GlobalOpenTelemetry.get();
+    OpenTelemetry openTelemetry = GlobalOpenTelemetry.get();
+    if (!(openTelemetry instanceof OpenTelemetrySdk)) {
+      throw new IllegalStateException(
+          "Trying to access global OpenTelemetrySdk but global OpenTelemetry is not an "
+              + "instance created by this SDK.");
+    }
+    return (OpenTelemetrySdk) openTelemetry;
   }
 
   /** Returns the global {@link SdkTracerManagement}. */
   public static SdkTracerManagement getGlobalTracerManagement() {
     TracerProvider tracerProvider = GlobalOpenTelemetry.get().getTracerProvider();
     if (!(tracerProvider instanceof ObfuscatedTracerProvider)) {
       throw new IllegalStateException(
-          "Trying to access global TracerSdkManagement but global TracerProvider is not an "
+          "Trying to access global SdkTracerManagement but global TracerProvider is not an "
               + "instance created by this SDK.");
     }
-    return (SdkTracerProvider) ((ObfuscatedTracerProvider) tracerProvider).unobfuscate();
+    return ((ObfuscatedTracerProvider) tracerProvider).unobfuscate();
```

Comment:
I'm surprised. Why is it possible to remove this cast here?

Reply:
This PR changes the `unobfuscate()` method to return an SdkTracerProvider, rather than just `TracerProvider`

Reply:
yes :) because now I can enforce that Builder only accepts SDK instances :)